client/websocket/websocket.go

```diff
@@ -264,35 +277,71 @@ out:
 // wsLoadMarket is the handler for the 'loadmarket' websocket route. Subscribes
 // the client to the notification feed and sends the order book.
 func wsLoadMarket(s *Server, cl *wsClient, msg *msgjson.Message) *msgjson.Error {
-	market := new(marketLoad)
-	err := json.Unmarshal(msg.Payload, market)
+	req := new(marketLoad)
+	err := json.Unmarshal(msg.Payload, req)
 	if err != nil {
 		errMsg := fmt.Sprintf("error unmarshalling marketload payload: %v", err)
 		s.log.Errorf(errMsg)
 		return msgjson.NewError(msgjson.RPCInternal, errMsg)
 	}
+	_, msgErr := loadMarket(s, cl, req)
+	return msgErr
+}
 
-	name, err := dex.MarketName(market.Base, market.Quote)
+func loadMarket(s *Server, cl *wsClient, req *marketLoad) (*bookFeed, *msgjson.Error) {
+	name, err := dex.MarketName(req.Base, req.Quote)
 	if err != nil {
 		errMsg := fmt.Sprintf("unknown market: %v", err)
 		s.log.Errorf(errMsg)
-		return msgjson.NewError(msgjson.UnknownMarketError, errMsg)
+		return nil, msgjson.NewError(msgjson.UnknownMarketError, errMsg)
 	}
 
-	feed, err := s.core.SyncBook(market.Host, market.Base, market.Quote)
+	feed, err := s.core.SyncBook(req.Host, req.Base, req.Quote)
 	if err != nil {
 		errMsg := fmt.Sprintf("error getting order feed: %v", err)
 		s.log.Errorf(errMsg)
-		return msgjson.NewError(msgjson.RPCOrderBookError, errMsg)
+		return nil, msgjson.NewError(msgjson.RPCOrderBookError, errMsg)
 	}
 
-	cl.feedLoopMtx.Lock()
-	if cl.feedLoop != nil {
-		cl.feedLoop.Stop()
-		cl.feedLoop.WaitForShutdown()
+	cl.feedMtx.Lock()
+	defer cl.feedMtx.Unlock()
+	if cl.feed != nil {
+		cl.feed.loop.Stop()
+		cl.feed.loop.WaitForShutdown()
+	}
+	cl.feed = &bookFeed{
+		BookFeed: feed,
+		loop:     newMarketSyncer(cl, feed, s.log.SubLogger(name)),
+		host:     req.Host,
+		base:     req.Base,
+		quote:    req.Quote,
+	}
+
+	return cl.feed, nil
+}
+
+func wsLoadCandles(s *Server, cl *wsClient, msg *msgjson.Message) *msgjson.Error {
+	req := new(candlesLoad)
+	err := json.Unmarshal(msg.Payload, req)
+	if err != nil {
+		errMsg := fmt.Sprintf("error unmarshalling candlesLoad payload: %v", err)
+		s.log.Errorf(errMsg)
+		return msgjson.NewError(msgjson.RPCInternal, errMsg)
+	}
+	cl.feedMtx.RLock()
+	feed := cl.feed
```

Comment:
After taking another look, I'm wondering if there's anyway a `"loadcandles"` could happen before `"loadmarket"`, which makes the feed.  Should `wsLoadCandles` either error if not already subscribed (with a non-nil feed) or even do a `loadMarket` with the embedded `candlesLoad.marketLoad`?  Just error maybe?